tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15541,46 +15541,47 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, llvm::StringRef StageName,
       S.Diags.Report(NodeDG->getLocation(), diag::note_defined_here)
           << NodeDG->getSpelling();
     }
-    // Node with NodeMaxDispatchGrid must have SV_DispatchGrid semantic.
-    if (NodeMDG) {
-      bool Found = false;
-      for (FunctionDecl::param_iterator I = FD->param_begin(),
-                                        E = FD->param_end();
-           I != E; ++I) {
-        QualType ParamType = (*I)->getType().getCanonicalType();
-
-        // Find parameter that is the node input record
-        if (hlsl::IsHLSLNodeInputType(ParamType)) {
-          // Node input records are template types
-          if (const RecordType *NodeInputRT = dyn_cast<RecordType>(ParamType)) {
-            if (const ClassTemplateSpecializationDecl *templateDecl =
-                    dyn_cast<ClassTemplateSpecializationDecl>(
-                        NodeInputRT->getDecl())) {
-
-              // Get the input record struct
-              auto &TemplateArgs = templateDecl->getTemplateArgs();
-              DXASSERT_NOMSG(TemplateArgs.size() >= 1);
-              QualType Arg0Type = TemplateArgs.get(0).getAsType();
-              const RecordType *NodeInputStructType =
-                  Arg0Type->getAsStructureType();
-              if (nullptr != NodeInputStructType) {
-                CXXRecordDecl *NodeInputStructDecl =
-                    dyn_cast<CXXRecordDecl>(NodeInputStructType->getDecl());
-                if (nullptr != NodeInputStructDecl) {
-                  // Make sure there is exactly one SV_DispatchGrid semantics
-                  // and it has correct type.
-                  DiagnoseDispatchGridSemantics(S, NodeInputStructDecl, Found);
-                }
+    // Diagnose dispatch grid semantics.
+    bool Found = false;
+    for (FunctionDecl::param_iterator I = FD->param_begin(),
+                                      E = FD->param_end();
+         I != E; ++I) {
+      QualType ParamType = (*I)->getType().getCanonicalType();
+
+      // Find parameter that is the node input record
+      if (hlsl::IsHLSLNodeType(ParamType)) {
+        // Node records are template types
+        if (const RecordType *NodeRT = dyn_cast<RecordType>(ParamType)) {
+          if (const ClassTemplateSpecializationDecl *templateDecl =
+                  dyn_cast<ClassTemplateSpecializationDecl>(
+                      NodeRT->getDecl())) {
+
+            // Get the record struct
+            auto &TemplateArgs = templateDecl->getTemplateArgs();
+            DXASSERT_NOMSG(TemplateArgs.size() >= 1);
+            QualType Arg0Type = TemplateArgs.get(0).getAsType();
+            if (const RecordType *NodeStructType =
+                    Arg0Type->getAsStructureType()) {
+              if (CXXRecordDecl *NodeStructDecl =
+                      dyn_cast<CXXRecordDecl>(NodeStructType->getDecl())) {
```

Comment:
As I commented on another PR, getting the record type from a `NodeOutput` or `NodeInputRecord` should be a helper function in HLSLTypes.cpp.  That would greatly simplify this code.

Reply:
Fixed.